crm_multicompany_reporting_currency/models/crm_lead.py

```diff
@@ -51,14 +51,20 @@ def _compute_multicompany_reporting_currency_id(self):
     def _compute_currency_rate(self):
         # similar to currency_rate on sale.order
         for record in self:
-            if (
+            date = record.create_date or fields.Date.today()
+            if not record.company_id:
+                record.currency_rate = (
+                    record.company_currency.with_context(date=date).rate or 1.0
+                )
+                continue
```

Comment:
`continue` is useless as rest of the function is under condition `elif` and `else`.
```suggestion
```